.werft/jobs/build/deploy-to-preview-environment.ts

```diff
@@ -153,6 +153,14 @@ export async function deployToPreviewEnvironment(werft: Werft, jobConfig: JobCon
     }
 
     werft.phase(phases.PREDEPLOY, "Checking for existing installations...");
+
+    // Because of a workspace label the branch name that can be used to create a preview environment is limited to 20 chars
+    // echo -n "gitpod.io/registry-facade_ready_ns_staging-" | wc -c
+    const maxBranchNameLength = 20;
+    if (deploymentConfig.destname.length > maxBranchNameLength) {
+        werft.fail(phases.PREDEPLOY, `The branch name ${deploymentConfig.destname} is more than ${maxBranchNameLength} character. Please choose a shorter name!`)
```

Comment:
We should fail much sooner than that - best before we do anything.
Note: don't forget to tie this to the absence of `no-preview`

Reply:
:100:

Reply:
I put it there as it is the phase where it would break. I created an extra phase that fails early. I tested it with this branch [[1](https://werft.gitpod-dev.com/job/gitpod-build-wulfthimm-max-branch-name-lenght-is-toooooooo.18)], with `noPreview` [[2](https://werft.gitpod-dev.com/job/gitpod-build-wulfthimm-max-branch-name-lenght-is-toooooooo.17)] and with a branch that has a shorter name [[3](https://werft.gitpod-dev.com/job/gitpod-build-wth-test.65)].